### ISSUE TITLE
Telesync plugin behaves poorly by default w/no config

### ISSUE THREAD

**Issue by pleasantone** [hangoutsbot/hangoutsbot#537]
Lots of decode failures and other crap, over and over and over again.
By default, if there is no configuration, even if the plugin is loaded, it should be dormant.
Some of us enable all plugins for testing.

This is after a pip3 install -r requirements.txt --upgrade too...

```
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: data = yield from response.json()
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/aiohttp/cli
Mar 09 12:18:10 hangupsbot[29722]: return loads(stripped.decode(encoding))
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/__init__.py", line 319, in
Mar 09 12:18:10 hangupsbot[29722]: return _default_decoder.decode(s)
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/decoder.py", line 339, in
Mar 09 12:18:10 hangupsbot[29722]: obj, end = self.raw_decode(s, idx=_w(s, 0).end())
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/lib/python3.5/json/decoder.py", line 357, in
Mar 09 12:18:10 hangupsbot[29722]: raise JSONDecodeError("Expecting value", s, err.value) f
Mar 09 12:18:10 hangupsbot[29722]: json.decoder.JSONDecodeError: Expecting value: line 1 co
Mar 09 12:18:10 hangupsbot[29722]: During handling of the above exception, another exceptio
Mar 09 12:18:10 hangupsbot[29722]: Traceback (most recent call last):
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: result = yield from self.getUpdates(offset=offset, timeo
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: return (yield from self._parse(r))
Mar 09 12:18:10 hangupsbot[29722]: File "/usr/local/lib/python3.5/dist-packages/telepot/asy
Mar 09 12:18:10 hangupsbot[29722]: raise BadHTTPResponse(response.status, text)
Mar 09 12:18:10 hangupsbot[29722]: telepot.exception.BadHTTPResponse: (200, '<!DOCTYPE html
Mar 09 12:18:10 hangupsbot[29722]: Bots are special Telegram accounts designed to handle me
Mar 09 12:18:10 hangupsbot[29722]: t is that you don&#39;t need to know anything about how
Mar 09 12:18:10 hangupsbot[29722]: links or username search to find your bot.</li>\n<li>Bot
Mar 09 12:18:10 hangupsbot[29722]: " style="max-height: 300px; padding: 10px 5px" /></a>\n\
```

**Comment by makiftasova** [comment]
**TL;DR**
 Telegram Bot needs an API key from [BotFather](https://telegram.me/BotFather) account. and when you set 'telegram_bot_api_key' to Telegram Bot API key in config.json, this error should be fixed.

Also I'm going to push a little changes for this issue ( you can find updated version at my branch [makiftasova/hangoutsbot](https://github.com/makiftasova/hangoutsbot/tree/staging)).

**Long Story**
telesync plugin depends on [telepot](https://github.com/nickoala/telepot), and when you initialize bot, telesync gets Telegram Bot API key from config.json, pass the API key to telepot. If given API key is not valid, telepot gets whole html page of [Bots: An introduction for developers](https://core.telegram.org/bots) page as response from Telegram servers and can't parse it. Thus causes telepot to dump all of the html page into logs.

**Comment by pleasantone** [comment]
It should not start without configuration explicitly enabling it.

**Comment by makiftasova** [comment]
Could you try commit [2e50f70](https://github.com/makiftasova/hangoutsbot/commit/2e50f7074b4a4176481e0633202d4b3f8afb5446)

added a "enabled" switch in config.json which should control bot's enable/disable status explicitly. also collected all telesync related stuff into one dictionary key

**Comment by pleasantone** [comment]
That solved the initialization problem, but what we really should be doing is not running anything at all (including modifying the config file) if the config is not there.  I may rewrite that bit of code.

**Comment by makiftasova** [comment]
It will be much appreciated if you can do that. 

Or I think I'll have some time in near future (1 or 2 weeks) to update telesync. I'll add this issue to my to do list

**Comment by Terrance** [comment]
In v3, telesync will return immediately if its config isn't present or is set to disabled, and will fail fast if the API key is missing.
